wandb/filesync/dir_watcher.py

```diff
@@ -223,7 +223,15 @@ def emitter(self) -> Optional["wd_api.EventEmitter"]:
             return None
 
     def update_policy(self, path: GlobStr, policy: "PolicyName") -> None:
-        if path == glob.escape(path):
+        # When we're dealing with one of our own media files, there's no need
+        # to store the policy in memory.  _get_file_event_handler will always
+        # return PolicyNow.  Using the path makes syncing historic runs much
+        # faster if the name happens to include glob escapable characters.  In
+        # the future we may add a flag to "files" records that indicates it's
+        # policy is not dynamic and doesn't need to be stored / checked.
+        if path.startswith("media/"):
+            pass
+        elif path == glob.escape(path):
             save_name = SaveName(
                 os.path.relpath(os.path.join(self._dir, path), self._dir)
             )
```

Comment:
I think the new branch needs to check `save_name.startswith`, rather than `path.startswith`:
```suggestion
        save_name = SaveName(
            os.path.relpath(os.path.join(self._dir, path), self._dir)
        )
        if save_name.startswith("media/"):
            pass
        elif path == glob.escape(path):
```
Reason 1: `save_name` is guaranteed to be relative to the run-files directory, whereas I think `path` might sometimes be absolute (judging by this `relpath(join(d, path), d)` pattern).

Reason 2: consistency with `_get_file_event_handler` below.

Reply:
Great catch.